src/main/python/integratedPython.ts

```diff
@@ -61,40 +67,49 @@ export const getIntegratedPython = async (
     onProgress: (percentage: number, stage: 'download' | 'extract') => void
 ): Promise<PythonInfo> => {
     const platform = getPlatform();
-    const pythonPath = path.resolve(path.join(directory, getExecutableRelativePath(platform)));
+    const { url, version, path: relativePath } = downloads[platform];
 
+    const pythonPath = path.resolve(path.join(directory, relativePath));
     const pythonBinExists = await checkFileExists(pythonPath);
 
-    if (!pythonBinExists) {
-        log.info(`Integrated Python not found at ${pythonPath}`);
+    if (pythonBinExists) {
+        const pythonInfo = await checkPythonPaths([pythonPath]);
+        if (eq(pythonInfo.version, version)) {
```

Comment:
Minor thing: Could you `import semver from 'semver';`, so that this line reads `semver.eq`? I think that would be more readable.